internal/backends/winit/winitwindowadapter.rs

```diff
@@ -719,3 +725,28 @@ impl Default for WindowProperties {
         Self { scale_factor: Property::new(1.0) }
     }
 }
+
+// Winit doesn't automatically resize the window to satisfy constraints. Qt does it though, and so do we here.
+fn adjust_window_size_to_satisfy_constraints(
+    winit_window: &winit::window::Window,
+    min_size: Option<winit::dpi::PhysicalSize<f32>>,
+    max_size: Option<winit::dpi::PhysicalSize<f32>>,
+) {
+    let mut window_size = winit_window.inner_size();
+
+    if let Some(min_size) = min_size {
+        let min_size = min_size.cast();
+        window_size.width = window_size.width.max(min_size.width);
+        window_size.height = window_size.height.max(min_size.height);
+    }
+
+    if let Some(max_size) = max_size {
+        let max_size = max_size.cast();
+        window_size.width = window_size.width.min(max_size.width);
+        window_size.height = window_size.height.min(max_size.height);
+    }
+
+    if window_size != winit_window.inner_size() {
+        let _ = winit_window.request_inner_size(window_size);
```

Comment:
I think we might need to do something with the return value here.
Other calls to this function store the value. Also we might need to dispatch a resize event

Reply:
Now, I change to dispatch a resize event instead of resizing directly.
Is this a good solution?